### ISSUE TITLE
Extend result structure

### ISSUE THREAD

**Issue by arnegevaert** [zoeparman/benchmark#129]
Result object should support **nested metrics.**
This allows for:
- softmax/logit versions without re-running benchmark
- correlation/MSE infidelity without re-running benchmark

Possible layout:
```
- deletion
  - logit
    - InputXGradient [128 x 50]
    - IntegratedGradients [128 x 50]
  - softmax
    - InputXGradient [128 x 50]
    - IntegratedGradients [128 x 50]
  - sigmoid
    - InputXGradient [128 x 50]
    - IntegratedGradients [128 x 50]
- infidelity
  - mse
    - InputXGradient [128 x 1]
    - IntegratedGradients [128 x 1]
  - corr
    - InputXGradient [128 x 1]
    - IntegratedGradients [128 x 1]
```

Also add metadata flag to indicate if results are **"higher/lower = better"** (`inverted: bool`)
